nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py

```diff
@@ -440,17 +498,242 @@ def build_train_valid_test_datasets(self):
+
+            if cfg.concat_sampling_probabilities is None or not isinstance(
+                cfg.concat_sampling_probabilities, ListConfig
+            ):
+                raise ValueError(
+                    f"concat_sampling_probabilities must be a ListConfig with the same number of files in src_file_name and tgt_file_name, found {cfg.concat_sampling_probabilities}"
+                )
+            if len(cfg.concat_sampling_probabilities) != len(cfg.src_file_name):
+                raise ValueError(
+                    f"concat_sampling_probabilities must be of the same size as src_file_name and tgt_file_name. Provided size {len(cfg.concat_sampling_probabilities)}, number of datasets {len(cfg.src_file_name)}"
+                )
+
+            datasets = []
+            for src_file, tgt_fille in zip(cfg.src_file_name, cfg.tgt_file_name):
+                dataset = MegatronSequenceToSequenceDataset(
+                    src_dataset_prefix=src_file,
+                    tgt_dataset_prefix=tgt_fille,
+                    src_tokenizer=self.encoder_tokenizer,
+                    tgt_tokenizer=self.decoder_tokenizer,
+                    max_encoder_seq_length=cfg.max_seq_length,
+                    max_decoder_seq_length=cfg.max_seq_length,
+                    start_index=0,
+                    end_index=None,
+                    data_impl="mmap",
+                    skip_warmup=True,
+                    dataset_index_sampling_override=False,
+                )
+                datasets.append(dataset)
+            dataset = BlendableDataset(datasets=datasets, weights=cfg.concat_sampling_probabilities)
+        else:
+            dataset = MegatronSequenceToSequenceDataset(
+                src_dataset_prefix=cfg.src_file_name,
+                tgt_dataset_prefix=cfg.tgt_file_name,
+                src_tokenizer=self.encoder_tokenizer,
+                tgt_tokenizer=self.decoder_tokenizer,
+                max_encoder_seq_length=cfg.max_seq_length,
+                max_decoder_seq_length=cfg.max_seq_length,
+                start_index=0,
+                end_index=None,
+                data_impl="mmap",
+                skip_warmup=True,
+                dataset_index_sampling_override=False,
+            )
+        return dataset
+
+    def build_text_memmap_dataset_from_config(self, cfg: DictConfig):
+        pass
+
+    def build_tarred_train_dataset(self):
+        return MTEncDecModel._setup_dataset_from_config(
+            cfg=self._cfg.train_ds,
+            encoder_tokenizer=self.encoder_tokenizer,
+            decoder_tokenizer=self.decoder_tokenizer,
+            global_rank=parallel_state.get_data_parallel_rank(),
+            world_size=parallel_state.get_data_parallel_world_size(),
+            multilingual=self.multilingual,
+            multilingual_ids=self.multilingual_ids,
+        )
+
+    def build_megatron_train_valid_test_datasets(self):
```

Comment:
Isn't this method builds only train dataset? (i..e, why is it called `build_megatron_train_valid_test_datasets`?)